pollux/core/src/main/scala/org/hyperledger/identus/pollux/core/service/uriResolvers/DidUrlResolver.scala

```diff
@@ -1,17 +1,112 @@
 package org.hyperledger.identus.pollux.core.service.uriResolvers
 
-import org.hyperledger.identus.pollux.vc.jwt.DidResolver
+import io.lemonlabs.uri.{Url, UrlPath}
+import org.hyperledger.identus.pollux.vc.jwt
+import org.hyperledger.identus.pollux.vc.jwt.*
+import org.hyperledger.identus.shared.crypto.Sha256Hash
 import org.hyperledger.identus.shared.http.{GenericUriResolverError, UriResolver}
+import org.hyperledger.identus.shared.models.StatusCode
 import zio.*
 
 class DidUrlResolver(httpUrlResolver: HttpUrlResolver, didResolver: DidResolver) extends UriResolver {
+  import DidUrlResolver.*
+
   def resolve(uri: String): IO[GenericUriResolverError, String] = {
-    ??? // TODO: implement did URL resolver
+
+    for {
+      parsed <- ZIO.fromTry(Url.parseTry(uri)).mapError(_ => InvalidURI(uri))
+      maybeResourceService = parsed.query.param("resourceService")
+      maybeResourcePath = parsed.query.param("resourcePath")
+      maybeResourceHash = parsed.query.param("resourceHash")
```

Comment:
could we have these String values as `val`s? E.g.

```
val resourcePath =  ...
val resourceHash = ...
```

Reply:
This was mostly a tiny linting comment
I meant to give a name to the hard-coded String constants. It is usually a best practice to avoid literal values (e.g. strings, numbers, other hard-coded lists) in code.
In general, when one has code that uses literal values, many linters would point you to create constants for those values instead of using them directly
The reasons to prefer that over using the values directly are:
- The values tend to be used in more than one place, typing errors can lead to annoying debugging issues
- The values may change, having them in a single place help refactoring
- The values are not always descriptive, and a name can give a meaningful description

w.r.t yo `val` _outside_ the for expression, or constants inside it, I don't have a preference